spring-integration-core/src/main/java/org/springframework/integration/aop/MessagePublishingInterceptor.java

```diff
@@ -59,29 +60,31 @@
 
 	private final MessagingTemplate messagingTemplate = new MessagingTemplate();
 
-	private volatile PublisherMetadataSource metadataSource;
+	private final ParameterNameDiscoverer parameterNameDiscoverer = new LocalVariableTableParameterNameDiscoverer();
 
-	private volatile DestinationResolver<MessageChannel> channelResolver;
+	private final PublisherMetadataSource metadataSource;
 
-	private volatile BeanFactory beanFactory;
+	private DestinationResolver<MessageChannel> channelResolver;
 
-	private final ParameterNameDiscoverer parameterNameDiscoverer = new LocalVariableTableParameterNameDiscoverer();
+	private BeanFactory beanFactory;
 
-	private volatile MessageBuilderFactory messageBuilderFactory = new DefaultMessageBuilderFactory();
+	private MessageBuilderFactory messageBuilderFactory = new DefaultMessageBuilderFactory();
 
-	private volatile boolean messageBuilderFactorySet;
+	private boolean messageBuilderFactorySet;
 
-	private volatile String defaultChannelName;
+	private String defaultChannelName;
 
 	public MessagePublishingInterceptor(PublisherMetadataSource metadataSource) {
 		Assert.notNull(metadataSource, "metadataSource must not be null");
 		this.metadataSource = metadataSource;
 	}
 
-
+	/**
+	 * @param metadataSource the {@link PublisherMetadataSource} to use.
+	 * @deprecated since 5.2 in favor constructor argument.
+	 */
+	@Deprecated
```

Comment:
Well, I'm not sure what is that *tooling*, but I guess we have already much more components with similar ctor-based initialization and no one has complained so far.

Therefore I believe we are safe to remove it now.
Well, at least deprecate since it is a mandatory ctor option anyway, but someone may still would like to call a setter. Although that is already not so good to mutate this component at runtime.